### ISSUE TITLE
Workaround for repeatOp not working for 1D vector case

### ISSUE THREAD

**Issue by umalesTT** [tenstorrent/tt-mlir#2112]
When performing softmax on 1D vector via repeatOp workaround https://github.com/tenstorrent/tt-mlir/commit/3a0fd0ee8dddbe91c8bf659f2ef92d86ba0b8483 results turn out to be inf (which is not the case for tensors where all dimensions are > 1 - then softmax gives expected results).  Below are 2 examples and their respective TTIR graphs. 

Tensor (1,32) and axis = 1
```
module @jit_apply_softmax attributes {mhlo.num_partitions = 1 : i32, mhlo.num_replicas = 1 : i32} {
  func.func public @main(%arg0: tensor<1x32xf32>) -> (tensor<1x32xf32> {jax.result_info = ""}) {
    %0 = tensor.empty() : tensor<1xf32>
    %1 = "ttir.max"(%arg0, %0) <{dim_arg = [1 : i32], keep_dim = false}> : (tensor<1x32xf32>, tensor<1xf32>) -> tensor<1xf32>
    %2 = "ttir.constant"() <{value = dense<0xFF800000> : tensor<1xf32>}> : () -> tensor<1xf32>
    %3 = tensor.empty() : tensor<1xf32>
    %4 = "ttir.broadcast"(%2, %3) <{broadcast_dimensions = array<i64: 1>}> : (tensor<1xf32>, tensor<1xf32>) -> tensor<1xf32>
    %5 = tensor.empty() : tensor<1xf32>
    %6 = "ttir.maximum"(%4, %1, %5) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<1xf32>, tensor<1xf32>, tensor<1xf32>) -> tensor<1xf32>
    %7 = tensor.empty() : tensor<1x1xf32>
    %8 = "ttir.reshape"(%6, %7) <{shape = [1 : i32, 1 : i32]}> : (tensor<1xf32>, tensor<1x1xf32>) -> tensor<1x1xf32>
    %9 = tensor.empty() : tensor<1x1xf32>
    %10 = "ttir.broadcast"(%8, %9) <{broadcast_dimensions = array<i64: 1, 1>}> : (tensor<1x1xf32>, tensor<1x1xf32>) -> tensor<1x1xf32>
    %11 = tensor.empty() : tensor<1x32xf32>
    %12 = "ttir.broadcast"(%10, %11) <{broadcast_dimensions = array<i64: 1, 32>}> : (tensor<1x1xf32>, tensor<1x32xf32>) -> tensor<1x32xf32>
    %13 = tensor.empty() : tensor<1x32xf32>
    %14 = "ttir.subtract"(%arg0, %12, %13) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<1x32xf32>, tensor<1x32xf32>, tensor<1x32xf32>) -> tensor<1x32xf32>
    %15 = tensor.empty() : tensor<1x32xf32>
    %16 = "ttir.exp"(%14, %15) <{operandSegmentSizes = array<i32: 1, 1>}> : (tensor<1x32xf32>, tensor<1x32xf32>) -> tensor<1x32xf32>
    %17 = tensor.empty() : tensor<1xf32>
    %18 = "ttir.sum"(%16, %17) <{dim_arg = [1 : i32], keep_dim = false}> : (tensor<1x32xf32>, tensor<1xf32>) -> tensor<1xf32>
    %19 = tensor.empty() : tensor<1x1xf32>
    %20 = "ttir.reshape"(%18, %19) <{shape = [1 : i32, 1 : i32]}> : (tensor<1xf32>, tensor<1x1xf32>) -> tensor<1x1xf32>
    %21 = tensor.empty() : tensor<1x1xf32>
    %22 = "ttir.broadcast"(%20, %21) <{broadcast_dimensions = array<i64: 1, 1>}> : (tensor<1x1xf32>, tensor<1x1xf32>) -> tensor<1x1xf32>
    %23 = tensor.empty() : tensor<1x32xf32>
    %24 = "ttir.broadcast"(%22, %23) <{broadcast_dimensions = array<i64: 1, 32>}> : (tensor<1x1xf32>, tensor<1x32xf32>) -> tensor<1x32xf32>
    %25 = tensor.empty() : tensor<1x32xf32>
    %26 = "ttir.div"(%16, %24, %25) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<1x32xf32>, tensor<1x32xf32>, tensor<1x32xf32>) -> tensor<1x32xf32>
    return %26 : tensor<1x32xf32>
  }
}
```
Tensor (32,) and axis = 0
```
module @jit_apply_softmax attributes {mhlo.num_partitions = 1 : i32, mhlo.num_replicas = 1 : i32} {
  func.func public @main(%arg0: tensor<32xf32>) -> (tensor<32xf32> {jax.result_info = ""}) {
    %0 = tensor.empty() : tensor<1xf32>
    %1 = "ttir.max"(%arg0, %0) <{dim_arg = [0 : i32], keep_dim = false}> : (tensor<32xf32>, tensor<1xf32>) -> tensor<1xf32>
    %2 = "ttir.constant"() <{value = dense<0xFF800000> : tensor<1xf32>}> : () -> tensor<1xf32>
    %3 = tensor.empty() : tensor<1xf32>
    %4 = "ttir.maximum"(%2, %1, %3) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<1xf32>, tensor<1xf32>, tensor<1xf32>) -> tensor<1xf32>
    %5 = tensor.empty() : tensor<1xf32>
    %6 = "ttir.broadcast"(%4, %5) <{broadcast_dimensions = array<i64: 1>}> : (tensor<1xf32>, tensor<1xf32>) -> tensor<1xf32>
    %7 = tensor.empty() : tensor<32xf32>
    %8 = "ttir.broadcast"(%6, %7) <{broadcast_dimensions = array<i64: 32>}> : (tensor<1xf32>, tensor<32xf32>) -> tensor<32xf32>
    %9 = tensor.empty() : tensor<32xf32>
    %10 = "ttir.subtract"(%arg0, %8, %9) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<32xf32>, tensor<32xf32>, tensor<32xf32>) -> tensor<32xf32>
    %11 = tensor.empty() : tensor<32xf32>
    %12 = "ttir.exp"(%10, %11) <{operandSegmentSizes = array<i32: 1, 1>}> : (tensor<32xf32>, tensor<32xf32>) -> tensor<32xf32>
    %13 = tensor.empty() : tensor<1xf32>
    %14 = "ttir.sum"(%12, %13) <{dim_arg = [0 : i32], keep_dim = false}> : (tensor<32xf32>, tensor<1xf32>) -> tensor<1xf32>
    %15 = tensor.empty() : tensor<1xf32>
    %16 = "ttir.broadcast"(%14, %15) <{broadcast_dimensions = array<i64: 1>}> : (tensor<1xf32>, tensor<1xf32>) -> tensor<1xf32>
    %17 = tensor.empty() : tensor<32xf32>
    %18 = "ttir.broadcast"(%16, %17) <{broadcast_dimensions = array<i64: 32>}> : (tensor<1xf32>, tensor<32xf32>) -> tensor<32xf32>
    %19 = tensor.empty() : tensor<32xf32>
    %20 = "ttir.div"(%12, %18, %19) <{operandSegmentSizes = array<i32: 2, 1>}> : (tensor<32xf32>, tensor<32xf32>, tensor<32xf32>) -> tensor<32xf32>
    return %20 : tensor<32xf32>
  }
}

```
Once https://github.com/tenstorrent/tt-metal/pull/17270 is merged, there shouldn't be need for this workaround, but we should keep in mind this flaw until then.

**Comment by sdjordjevicTT** [comment]
@umalesTT, thanks for filing this issue. Someone will take a look at it.
